### PR TITLE
GCW 3490 enhance shared offer feature

### PR DESCRIPTION
**Pull request by shreyas098**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-3490

### What does this PR do?

FEATURE: Enhance shared offers

### CODE REVIEW

File: app/models/concerns/offer_filtering.rb [hunk 1]
```diff
@@ -72,6 +73,11 @@ def self.last_6pm
       t
     end
 
+    def self.shareable
+      res = joins("INNER JOIN shareables ON shareables.resource_id=offers.id")
+      res
```
Comment (bharat619) [inline app/models/concerns/offer_filtering.rb:78]:
```suggestion
       joins("INNER JOIN shareables ON shareables.resource_id=offers.id")
```

File: spec/controllers/api/v1/offers_controller_spec.rb [hunk 2]
```diff
@@ -477,9 +477,30 @@
...
+    let(:shareable1) {create :shareable, resource: reviewing_offer}
+    let(:shareable2) {create :shareable, resource: submitted_offer}
     before(:each) { generate_and_set_token(reviewer) }
     subject { JSON.parse(response.body) }
 
+    context "Shareable Offers/Published offers filter" do
+      before(:each){
+        shareable1
+        shareable2
+      }
```
Comment (bharat619) [inline spec/controllers/api/v1/offers_controller_spec.rb:489]: Remove this and use `let!`, which creates the db record instantly

let!(:shareable1) {create :shareable, resource: reviewing_offer}
let!(:shareable2) {create :shareable, resource: submitted_offer}

File: spec/controllers/api/v1/offers_controller_spec.rb [hunk 3]
```diff
@@ -480,6 +480,23 @@
...
+      it "returns offers published on charity site" do
+        get :search, params: { shareable: true }
+        expect(response.status).to eq(200)
+        expect(subject['offers'].size).to eq(2)
+      end
+
+      it "returns offers published on charity site of particular state" do
+        get :search, params: { shareable: true, state: "submitted" }
+        expect(response.status).to eq(200)
+        expect(subject['offers'].size).to eq(1)
```
Comment (swatijadhav) [inline spec/controllers/api/v1/offers_controller_spec.rb:496]: Along with checking count, you can also verify if the correct offer is returned in the response or not.

Comment (shreyas098) [reply spec/controllers/api/v1/offers_controller_spec.rb:496]: sure..

**Review by swatijadhav — APPROVED** [review]

**Review by bharat619 — APPROVED** [review]

**Pull request merged**